pkg/controller/account/byoc.go

```diff
@@ -190,7 +202,7 @@ func CreateRole(reqLogger logr.Logger, byocRole string, principalARN string, byo
 			Effect: "Allow",
 			Action: []string{"sts:AssumeRole"},
 			Principal: &awsv1alpha1.Principal{
-				AWS: principalARN,
+				AWS: []string{principalARN, SREAccessARN},
```

Comment:
Any reason to pass this all the way through instead of use the constant directly here?

Reply:
Mainly for when we add tests to this. If we try to pull the constant directly we'll have a hard time trying to build the tests if we need to change that constant for the test.

Reply:
The tests will have it in the configmap too right?

Reply:
They could, but that's an extra mock you have to do.  This way instead of having to mock out a configmap and add it to the mock-api for each test you'd just pass in either one.

I mean, I think I understand the concern, there are a lot of parameters to these functions to begin with, and I'm adding a new one.

What if instead of passing in two separate strings for the principal we change it so that we pass in a single `[]string` for principals.  Then we keep it to one single parameter for the principals but also have the ease of just flipping a variable in a test?

Reply:
Oh, I like that idea.